packages/amplify-e2e-tests/src/rds-v2-tests-common/rds-field-auth-lambda.ts

```diff
@@ -0,0 +1,620 @@
+        username,
+        password,
+        region,
+      };
+
+      const db = await setupRDSInstanceAndData(dbConfig, queries);
+      port = db.port;
+      host = db.endpoint;
+    };
+
+    const cleanupDatabase = async (): Promise<void> => {
+      await deleteDBInstance(identifier, region);
+    };
+
+    const initProjectAndImportSchema = async (): Promise<void> => {
+      await initJSProjectWithProfile(projRoot, {
+        disableAmplifyAppCreation: false,
+        name: projName,
+      });
+
+      const metaAfterInit = getProjectMeta(projRoot);
+      region = metaAfterInit.providers.awscloudformation.Region;
+      await setupDatabase();
+
+      const rdsSchemaFilePath = path.join(projRoot, 'amplify', 'backend', 'api', apiName, 'schema.sql.graphql');
+
+      await addApiWithAllAuthModes(projRoot, { transformerVersion: 2, apiName });
+
+      await importRDSDatabase(projRoot, {
+        database,
+        engine,
+        host,
+        port,
+        username,
+        password,
+        useVpc: true,
+        apiExists: true,
+      });
+
+      const schema = /* GraphQL */ `
+        input AMPLIFY {
+          engine: String = "${engineName}"
+        }
+
+        # Lambda Authorizer can't delete a record
+        # Lambda Authorizer can't update SSN field
+        # Lambda Authorizer can't read SSN field 
+        # Lambda Authorizer can create a record with all fields, but can't read SSN response
+        # Lambda Authorizer subscription can't read SSN field
+        type Person1 @model @auth(rules: [{allow: custom, provider: function}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: custom, provider: function, operations: [create]}])
+        }
+
+        # Lambda Authorizer can't create a record with SSN value
+        # Lambda Authorizer can create a record without a SSN value
+        # Lambda Authorizer can't update a record with SSN value
+        # Lambda Authorizer can't delete a record
+        type Person2 @model @auth(rules: [{allow: custom, provider: function}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: custom, provider: function, operations: [read]}])
+        }
+
+        # Lambda Authorizer can't create a record with SSN value
+        # Lambda Authorizer can delete a record
+        type Person3 @model @auth(rules: [{allow: custom, provider: function}, {allow: public, provider: iam}]) {
+          id: Int! @primaryKey
+          firstName: String
+          lastName: String
+          ssn: String @auth(rules: [{allow: public, provider: iam}, {allow: custom, provider: function, operations: [delete]}])
+        }
+      `;
+      writeFileSync(rdsSchemaFilePath, schema, 'utf8');
```

Comment:
missing test for update operation using custom function provider. Something that can be followed up.

Reply:
Will add in a follow up PR.